client/rest-high-level/src/main/java/org/elasticsearch/client/indices/CreateIndexResponse.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.indices;
+
+import org.elasticsearch.action.support.master.ShardsAcknowledgedResponse;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+
+/**
+ * A response for a create index action.
+ */
+public class CreateIndexResponse extends ShardsAcknowledgedResponse {
```

Comment:
Why do we need a separate response object in this case? As far as I see the parsing etc... is identical to  `org.elasticsearch.action.admin.indices.create.CreateIndexResponse`. I get that a copy of the request is useful to distiguish the type and typeless calls but the response looks similar in this case.

Reply:
I thought this would be most intuitive for users, because otherwise they'd be using a request from one package, and a response from another. It's also seemed like a step in the right long-term direction of having separate requests + responses for the HLRC.

Reply:
yea im ++ for this. im also ok w/ this just extending the server side create response until we can sever ties w server.

Reply:
If it's okay I'd like to keep it as is, since I think this gives the clearest API and isn't too bad in terms of code duplication.